libtenzir/src/tql2/eval_binary.cpp

```diff
@@ -255,11 +255,30 @@ struct BinOpKernel<Op, L, R> {
   }
 };
 
+template <>
+struct BinOpKernel<ast::binary_op::div, ip_type, int64_type> {
+  using result = subnet;
+
+  static auto evaluate(ip l, int64_t r) -> std::variant<result, const char*> {
+    // TODO: What if we use an IPv6 literal with the IPv4 prefix?
+    if (l.is_v4()) {
+      if (r < 0 or r > 32) {
+        return "subnet length out of range";
+      }
+      return subnet{l, detail::narrow_cast<uint8_t>(r + 96)};
+    }
+    if (r < 0 or r > 128) {
+      return "subnet length out of range";
+    }
+    return subnet{l, detail::narrow_cast<uint8_t>(r)};
+  }
```

Comment:
@dominiklohmann Thoughts regarding the todo?

Reply:
I think the comment points out a problem that cannot be fixed without having actual subnet literals. So the idea of having ip / int -> subnet we can probably discard.

Reply:
I'm thinking the same :( Well, too bad. But before I scrap this: What are our other options for constructing a subnet from a non-literal-IP? Or would we do we not see a strong need for this?

Reply:
Given that we do not have separate literals for IPv4 and IPv6 constructing it at runtime would require passing in which of the two it is. But I don't see a strong need for this currently, so let's avoid it for now.

Reply:
I think we can scrap this now, but we should extract all the other changes you made on the side into a separate PR.